org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/services/resourcePropertyServices.mtl

```diff
@@ -69,178 +69,215 @@ endif)
-				else
-					''
-				endif
-			endif
-		endif
-	endif
+    if (aProperty.occurs = ResourcePropertyOccurs::exactlyOne) then
+        '@OslcOccurs(Occurs.ExactlyOne)'
+    else
+        if (aProperty.occurs = ResourcePropertyOccurs::zeroOrOne) then
+            '@OslcOccurs(Occurs.ZeroOrOne)'
+        else
+            if (aProperty.occurs = ResourcePropertyOccurs::zeroOrMany) then
+                '@OslcOccurs(Occurs.ZeroOrMany)'
+            else
+                if (aProperty.occurs = ResourcePropertyOccurs::oneOrMany) then
+                    '@OslcOccurs(Occurs.OneOrMany)'
+                else
+                    ''
+                endif
+            endif
+        endif
+    endif
 else
-	''
+    ''
 endif)
 /]
 
 [query public valueTypeAnnotation(aProperty: ResourceProperty) : String =
 (if (not aProperty.valueType.oclIsUndefined()) then
-	if (hasOSLCSupportedValueType(aProperty)) then
-		'@OslcValueType(ValueType.'.concat(aProperty.valueType.toString()).concat(')')
-	else
-		''
-	endif
+    if (hasOSLCSupportedValueType(aProperty)) then
+        '@OslcValueType(ValueType.'.concat(aProperty.valueType.toString()).concat(')')
+    else
+        ''
+    endif
 else
-	''
+    ''
 endif)
 /]
 
 [comment  Either not supported by OSLC4J /]
 [query public representationAnnotation(aProperty: ResourceProperty) : String =
 (if (not aProperty.representation.oclIsUndefined()) then
-	if (not aProperty.representation.toString().equalsIgnoreCase('n_a')) then
-		if (not aProperty.representation.toString().equalsIgnoreCase('either')) then
-			'@OslcRepresentation(Representation.'.concat(aProperty.representation.toString().toUpperFirst()).concat(')')
-		else
-			''
-		endif
-	else
-		''
-	endif
+    if (not aProperty.representation.toString().equalsIgnoreCase('n_a')) then
+        if (not aProperty.representation.toString().equalsIgnoreCase('either')) then
+            '@OslcRepresentation(Representation.'.concat(aProperty.representation.toString().toUpperFirst()).concat(')')
+        else
+            ''
+        endif
+    else
+        ''
+    endif
+else
+    ''
+endif)
+/]
+
+[comment Only annotation if the rdfCollectionRepresentation & occurs properties are set.
+Also, only annotation if the occurs is "many"/]
+[query public rdfCollectionTypeAnnotation(aProperty: ResourceProperty) : String =
+(if ((not aProperty.rdfCollectionRepresentation.oclIsUndefined()) and (not aProperty.occurs.oclIsUndefined())) then
+    if (aProperty.occurs = ResourcePropertyOccurs::zeroOrMany or aProperty.occurs = ResourcePropertyOccurs::oneOrMany) then
+        if (aProperty.rdfCollectionRepresentation = ResourcePropertyRdfCollectionRepresentation::rdfList) then
+            '@OslcRdfCollectionType(collectionType = OslcRdfCollectionType.RDF_LIST)'
+        else
+            if (aProperty.rdfCollectionRepresentation = ResourcePropertyRdfCollectionRepresentation::rdfBag) then
+                '@OslcRdfCollectionType(collectionType = OslcRdfCollectionType.RDF_BAG)'
```

Comment:
stictly speaking, RDF List is an RDF Collection and RDF Bag or Seq are RDF _Containers_ but its OK.